1.data-exploration/scripts/nbconverted/2.test-train_split.py

```diff
@@ -87,73 +133,36 @@
 ].reset_index(drop=True)
 
 
-# In[8]:
+# In[39]:
 
 
 # split the data based on age category and sex
-train_df, test_df = train_test_split(
-    PA_effect_df, test_size=0.15, stratify=PA_effect_df.age_and_sex
+train_df, testandvalidation_df = train_test_split(
+    PA_effect_df, test_size=0.3, stratify=PA_effect_df.age_and_sex
 )
 train_df.reset_index(drop=True,inplace=True)
+testandvalidation_df.reset_index(drop=True,inplace=True)
+test_df, val_df = train_test_split(
+    testandvalidation_df, test_size=0.5
+)
 test_df.reset_index(drop=True,inplace=True)
+val_df.reset_index(drop=True,inplace=True)
 
 
-# In[9]:
-
-
-# preparing train dataframe to be scaled
-col_num = train_df.shape[1]
-train_scaled_df = train_df.iloc[:, 1:col_num-1]
-
-# scaling gene effect data to 0-1 range
-scaler = MinMaxScaler(feature_range=(0,1))
-train_scaled_df = scaler.fit_transform(train_scaled_df)
-
-# adding id column and age and sex column back
-train_scaled_df = pd.DataFrame(train_scaled_df)
-train_scaled_df.insert(0, train_df.columns[0], train_df[train_df.columns[0]])
-train_scaled_df.insert(col_num-1, train_df.columns[col_num-1], train_df[train_df.columns[col_num-1]])
-train_scaled_df.columns = train_df.columns
-train_scaled_df
-
-
-# In[10]:
-
-
-# preparing test dataframe to be scaled
-col_num = test_df.shape[1]
-test_scaled_df = test_df.iloc[:, 1:col_num-1]
-
-# scaling gene effect data to 0-1 range
-scaler = MinMaxScaler(feature_range=(0,1))
-test_scaled_df = scaler.fit_transform(test_scaled_df)
-
-# adding id column and age and sex column back
-test_scaled_df = pd.DataFrame(test_scaled_df)
-test_scaled_df.insert(0, test_df.columns[0], test_df[test_df.columns[0]])
-test_scaled_df.insert(col_num-1, test_df.columns[col_num-1], test_df[test_df.columns[col_num-1]])
-test_scaled_df.columns = test_df.columns
-test_scaled_df
-
-
-# In[11]:
+# In[40]:
 
 
-# save the TESTING dataframe
-test_df = test_df.reset_index(drop=True)
-testing_df_output = pathlib.Path("../0.data-download/data/VAE_test_df.csv")
-test_df.to_csv(testing_df_output, index=False)
-print(test_df.shape)
-test_df.head(3)
+#scale each dataframe
+train_scaled_df = scale_dataframe(train_df)
+test_scaled_df = scale_dataframe(test_df)
+val_scaled_df = scale_dataframe(val_df)
 
 
-# In[12]:
+# In[41]:
 
 
-# save the TRAINING dataframe
-train_df = train_df.reset_index(drop=True)
-training_df_output = pathlib.Path("../0.data-download/data/VAE_train_df.csv")
-train_df.to_csv(training_df_output, index=False)
-print(train_df.shape)
-train_df.head(3)
+#save each dataframe
+save_dataframe(train_scaled_df, "../0.data-download/data/VAE_train_df.csv")
+save_dataframe(test_scaled_df, "../0.data-download/data/VAE_test_df.csv")
+save_dataframe(val_scaled_df, "../0.data-download/data/VAE_val_df.csv")
```

Comment:
Consider making these resolved paths with `pathlib.Path("path/of/file").resolve()`